src/main/java/at/hannibal2/skyhanni/features/dungeon/DungeonFinderFeatures.kt

```diff
@@ -1,179 +1,280 @@
 import net.minecraftforge.fml.common.eventhandler.SubscribeEvent
+import kotlin.time.Duration
+import kotlin.time.Duration.Companion.milliseconds
 
 class DungeonFinderFeatures {
-
     private val config get() = SkyHanniMod.feature.dungeon.partyFinder
 
-    private val pricePattern = "([0-9]{2,3}K|[0-9]{1,3}M|[0-9]+\\.[0-9]M|[0-9] ?mil)".toRegex(RegexOption.IGNORE_CASE)
-    private val carryPattern = "(carry|cary|carries|caries|comp|to cata [0-9]{2})".toRegex(RegexOption.IGNORE_CASE)
-    private val nonPugPattern = "(perm|vc|discord)".toRegex(RegexOption.IGNORE_CASE)
-    private val memberPattern = "^ §.*?§.: §.([A-Z]+)§. \\(§.([0-9]+)§.\\)".toRegex(RegexOption.IGNORE_CASE)
-    private val ineligiblePattern =
-        "^§c(Requires .*$|You don't meet the requirement!|Complete previous floor first!$)".toRegex()
-    private val classLevelPattern = " §.(?<playerName>.*)§f: §e(?<className>.*)§b \\(§e(?<level>.*)§b\\)".toPattern()
-    private val notePattern = "^(§7§7Note: |§f[^§])".toRegex()
+    //  Repo group and patterns
+    private val repoGroup = RepoPattern.group("dungeon.finder.features")
+    private val pricePattern by repoGroup.pattern("price", "([0-9]{2,3}K|[0-9]{1,3}M|[0-9]+\\.[0-9]M|[0-9] ?MIL)")
+    private val carryPattern by repoGroup.pattern("carry", "(CARRY|CARY|CARRIES|CARIES|COMP|TO CATA [0-9]{2})")
+    private val nonPugPattern by repoGroup.pattern("nonpug", "(PERM|VC|DISCORD)")
+    private val memberPattern by repoGroup.pattern("member", ".*§.(?<playerName>.*)§f: §e(?<className>.*)§b \\(§e(?<level>.*)§b\\)")
+    private val ineligiblePattern by repoGroup.pattern(
+        "ineligible",
+        "^§c(Requires .*$|You don't meet the requirement!|Complete previous floor first!$)"
+    )
+    private val classLevelPattern by repoGroup.pattern(
+        "classlevel",
+        " §.(?<playerName>.*)§f: §e(?<className>.*)§b \\(§e(?<level>.*)§b\\)"
+    )
+    private val notePattern by repoGroup.pattern("note", "^(§7§7Note: |§f[^§])")
+    private val floorTypePattern by repoGroup.pattern("floortype", "(The Catacombs).*|.*(MM Catacombs).*")
+    private val checkIfPartyPattern by repoGroup.pattern("checkifparty", ".*('s Party)")
+    private val partyFinderTitlePattern by repoGroup.pattern("pfindertitle", "(Party Finder)")
+    private val catacombsGatePattern by repoGroup.pattern("catagate", "(Catacombs Gate)")
+    private val selectFloorPattern by repoGroup.pattern("selectfloor", "(Select Floor)")
+    private val entranceFloorPattern by repoGroup.pattern("entrance", "(.*Entrance)")
+    private val floorPattern by repoGroup.pattern("floor", "(Floor .*)")
+    private val anyFloorPattern by repoGroup.pattern("anyfloor", "(Any)")
+    private val masterModeFloorPattern by repoGroup.pattern("mmfloor", "(MM )|(.*Master Mode Catacombs)")
+    private val dungeonFloorPattern by repoGroup.pattern("dungeonfloor", "(Dungeon: .*)")
+    private val floorFloorPattern by repoGroup.pattern("floorpattern", "(Floor: .*)")
```

Comment:
all of these should be split over 4 lines each for better readability. Also you dont need to compact the pattern keys, feel free to type out the whole word and seperate multiple words with a `.` e.g. mmfloor -> floor.mastermode